rudder-core/src/main/scala/com/normation/rudder/domain/reports/ReportType.scala

```diff
@@ -49,66 +49,78 @@ sealed trait ReportType {
+  }
+  case object EnforceSuccess extends ReportType {
+    val severity = "Success"
+  }
+  case object EnforceRepaired extends ReportType{
+    val severity = "Repaired"
+  }
+  case object EnforceError extends ReportType{
+    val severity = "Error"
+  }
+
+  case object AuditNotApplicable extends ReportType {
+    val severity = "AuditNotApplicable"
+  }
+  case object AuditCompliant extends ReportType {
+    val severity = "Compliant"
+  }
+  case object AuditNonCompliant extends ReportType{
+    val severity = "NonCompliant"
+  }
+  case object AuditError extends ReportType{
+    val severity = "AuditError"
+  }
+
+  case object Unexpected extends ReportType{
+    val severity = "Unexpected"
+  }
+  case object NoAnswer extends ReportType{
+    val severity = "NoAnswer"
+  }
+  case object Disabled extends ReportType{
+    val severity = "ReportsDisabled"
+  }
+  case object Pending extends ReportType{
+    val severity = "Applying"
+  }
+  case object Missing extends ReportType{
+    val severity = "Missing"
+  }
 
   def getWorseType(reportTypes : Iterable[ReportType]) : ReportType = {
     if (reportTypes.isEmpty) {
-      NoAnswerReportType
+      NoAnswer
     } else {
-      ( reportTypes :\ (NotApplicableReportType : ReportType) ) {
-        case (_, UnexpectedReportType)     | (UnexpectedReportType, _)    => UnexpectedReportType
-        case (_, ErrorReportType)          | (ErrorReportType, _)         => ErrorReportType
-        case (_, RepairedReportType)       | (RepairedReportType, _)      => RepairedReportType
-        case (_, MissingReportType)        | (MissingReportType, _)       => MissingReportType
-        case (_, NoAnswerReportType)       | (NoAnswerReportType, _)      => NoAnswerReportType
-        case (_, DisabledReportType)| (DisabledReportType, _)      => DisabledReportType
-        case (_, PendingReportType)        | (PendingReportType, _)       => PendingReportType
-        case (_, SuccessReportType)        | (SuccessReportType, _)       => SuccessReportType
-        case (_, NotApplicableReportType)  | (NotApplicableReportType, _) => NotApplicableReportType
-        case _ => UnexpectedReportType
+      ( reportTypes :\ (EnforceNotApplicable : ReportType) ) {
+        case (_, Unexpected)          | (Unexpected, _)           => Unexpected
+        case (_, EnforceError)        | (EnforceError, _)         => EnforceError
+        case (_, EnforceRepaired)     | (EnforceRepaired, _)      => EnforceRepaired
+        case (_, Missing)             | (Missing, _)              => Missing
+        case (_, NoAnswer)            | (NoAnswer, _)             => NoAnswer
+        case (_, Disabled)            | (Disabled, _)             => Disabled
+        case (_, Pending)             | (Pending, _)              => Pending
+        case (_, EnforceSuccess)      | (EnforceSuccess, _)       => EnforceSuccess
+        case (_, EnforceNotApplicable)| (EnforceNotApplicable, _) => EnforceNotApplicable
```

Comment:
I'm not sure we want to keep the worst method, which is used only in the status report for a node, when there is several identical values. I think in that case, we just want to display the habitual compliance bar OR one not aggregate the different lines if they are of several types. 

And even if we keep it, I'm not sur about the semantic of having Compliant AND Success. It looks like something we don't want at all, because it must be an agent error in some way (on one node, having a multi-value for ONE directive giving both enforce and audit result... bad). 

Well, so for now, I didn't touched it.